Sources/BSON/JSON.swift

```diff
@@ -0,0 +1,138 @@
+import Foundation
+
+/// Enum representing a JSON value
+internal enum JSON: Codable {
+    case number(Double)
+    case string(String)
+    case bool(Bool)
+    indirect case array([JSON])
+    indirect case object([String: JSON])
+    case null
+
+    /// Initialize a `JSON` from a decoder
```

Comment:
as a general style thing, we try to write the docstrings in full sentences that end in periods. We don't follow this everywhere unfortunately, but in all new code we try to keep to it.

Reply:
fixed